src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -289,22 +293,30 @@ public static Version Version
                 version.wServicePackMajor = 0;
                 version.wServicePackMinor = 0;
 
-                return Win32Native.VerifyVersionInfoW(ref version,
+                isWindows8OrAbove = Win32Native.VerifyVersionInfoW(ref version,
                            Win32Native.VER_MAJORVERSION | Win32Native.VER_MINORVERSION | Win32Native.VER_SERVICEPACKMAJOR | Win32Native.VER_SERVICEPACKMINOR,
                            conditionMask);
             }
-        });
-        internal static bool IsWindows8OrAbove => s_IsWindows8OrAbove.Value;
+
+            return isWindows8OrAbove;
```

Comment:
Doesn’t get set here, so this is re-running every time in current.

Reply:
Good catch!